src/7bb60_len_41b0.c

```diff
@@ -356,7 +392,26 @@ void phys_update_interact_collider(void) {
 
 INCLUDE_ASM(s32, "7bb60_len_41b0", phys_check_interactable_collision);
 
-INCLUDE_ASM(s32, "7bb60_len_41b0", phys_can_player_interact);
+s32 phys_can_player_interact(void) {
+    PartnerActionStatus* partnerActionStatus = &gPartnerActionStatus;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    s32 ret = TRUE;
+
+    if (gPartnerActionStatus.actionState.b[0] != 0) {
+        if (gPartnerActionStatus.actionState.b[3] == 3) {
+            if (gPartnerActionStatus.actionState.b[0] < 3) {
+                ret = FALSE;
+            }
+        } else {
+            ret = FALSE;
+        }
+    } else if (!(gPlayerStatus.actionState == ACTION_STATE_IDLE ||
+                 gPlayerStatus.actionState == ACTION_STATE_WALK ||
+                 gPlayerStatus.actionState == ACTION_STATE_RUN)) {
+        ret = FALSE;
+    }
+    return ret;
```

Comment:
Does it work to `return FALSE` in each of the branches and `return TRUE` at the end (probably in an `else` to make flow clear)?

Reply:
No. In this file, 1 is stored into a register way early on in many of these funcs, which I don't know how to replicate without the way it currently is